cli/command_restore.go

```diff
@@ -0,0 +1,81 @@
+package cli
+
+import (
+	"context"
+	"io"
+	"os"
+	"path"
+
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/repo/object"
+	"github.com/kopia/kopia/snapshot/snapshotfs"
+)
+
+var (
+	restoreSnapshotID = restoreCommands.Arg("snapshot-id", "Snapshot ID from which to restore.").Required().String()
+	restoreTargetPath = restoreCommands.Arg("target-path", "Path to which the snapshot must be restored.").Required().String()
+)
+
+func runRestoreCommand(ctx context.Context, rep *repo.Repository) error {
+	oid, err := parseObjectID(ctx, rep, *restoreSnapshotID)
```

Comment:
Is `restoreSnapshotID` a snap id or the root id in the snapshot? Does `parseObjectID` or `snapshotfs.DirectoryEntry` handle both snapshot ids and directory ids??

Reply:
It is the `rootID` from the snapshot command

Reply:
Looks like the `snapshotfs.DirectoryEntry` doesn't validate the `objectID` before creating a directory entry and it fails when we do `Readdir`.
Will explore alternatives here